routes/profile/addUserProfile.js

```diff
@@ -0,0 +1,33 @@
+/*
+Purpose of API: This API will update the user's profile, and save their changes in the SQL database. 
```

Comment:
Right now I see the edit feature for the profile info is currently adding new rows instead of editing existing rows. In interest of db space, would you be able to switch it? I think you can use the edit Events api Saf made.

Reply:
double this

Reply:
You have two options for this:
1. Create a new API, `editUserProfile` and deal with the conditional logic of using `addUserProfile` and `editUserProfile` on the js side (RECOMMENDED)
2. Add conditionals to the API and make multiple queries. If the search query returns nothing, do an insert. If it returns a row, alter that row instead. You can refer to the `joinEvent` API on how to do this. (NOT RECOMMENDED)

Option 1 Is a better option for your case, as it's two different operations (create vs edit) and they should be separated into two APIs. On the front-end, you need to first check if the user has a user profile using your `getUserProfile` API. If they don't, make sure the button rendered is "Create Profile" and the handler uses the `addUserProfile` API. If they do, make sure the button rendered is "Edit Profile", and the handler uses the `editUserProfile` API.

Reply:
Thank you guys, this was really helpful!